instructure/canvas-android
# [All][MBL-13129] Use Refresh Tokens

All 401s will go through the CanvasAuthenticator from now on (aside from 'accounts/self' calls, since they frequently return 401 when we check if the user can masquerade). This will handle any authentication refreshes or necessary log outs.


To test:
 - Check mobile dev developer key for the one created by Nate (login as site admin, click on Developer Keys in the menu)
 - Set the client id/secret using the number above the 'Show Key' button, and the string shown after pressing the 'Show Key' button respectively (I set their ApiPrefs right in CanvasAuthenticator, before it goes to make the refresh call)
 - Log in to a user on mobile dev on the app in test
 - Log into web as that user, check for a token that has an expiration time on it
 - Wait 1 hour
 - See if the app still works for that user (navigate around the app)
 - Check on web if that tokens expiration date has been updated

You probably also want to test pre-existing users to make sure that all works, since we will be using their infinite token until they log out and log back in (I tested Student and Teacher, both seemed fine).

GOOD LUCK
